//python -m grpc_tools.protoc -I protos --python_out=grpc_generated_files --grpc_python_out=grpc_generated_files protos/*.proto
syntax = "proto3";
package auth;
//import "users.proto";
import "google/protobuf/wrappers.proto";


message User {
  string id = 1;
  string email = 2;
  string phone = 3;
  string last_name = 4;
  string first_name = 5;
  string middle_name = 6;
}

message RegisterUserRequest {
  string password = 1;
  string email = 2;
  string phone = 3;
  string last_name = 4;
  string first_name = 5;
  string middle_name = 6;
}

message RegisterUserResponse {
  User success_payload = 1;
  string error_payload = 2;
  bool error = 3;
  int32 status_code = 4;
}

message CheckUsernameRequest {
  string username = 1;
}

message LoginRequest {
  string email = 1;
  string password = 2;
}

message TokenSchema {
  string access_token = 1;
  string refresh_token = 2;
}

message LoginResponse {
  TokenSchema success_payload = 1;
  string error_payload = 2;
  bool error = 3;
  int32 status_code = 4;
}

message RefreshTokenRequest {
  string refresh_token = 1;
}


message RefreshTokenResponse {
  TokenSchema success_payload = 1;
  string error_payload = 2;
  bool error = 3;
  int32 status_code = 4;
}

message ValidateTokenRequest {
  string access_token = 1;
}

message ValidateTokenResponse {
  User success_payload = 1;
  string error_payload = 2;
  bool error = 3;
  int32 status_code = 4;
}

service AuthService {
  rpc register(RegisterUserRequest) returns (RegisterUserResponse);
  rpc login(LoginRequest) returns (LoginResponse);
  rpc refresh_token(RefreshTokenRequest) returns (RefreshTokenResponse);
  rpc validate_access_token(ValidateTokenRequest) returns (ValidateTokenResponse);
  rpc check_username (CheckUsernameRequest) returns (google.protobuf.BoolValue);
}


//message RegisterUserRequest {
//  string username = 1;
//  string password = 2;
//  string email = 3;
//}

//message  CreateUserResponse {
//  int32 id = 1;
//  string username = 2;
//  string email = 3;
//}


//
//message CheckUsernameResponse {
//  bool is_exists = 1;
//    }
//
//service UserService {
////  rpc create_user (CreateUserRequest) returns (CreateUserResponse);

//  rpc check_username (CheckUsernameRequest) returns (google.protobuf.BoolValue);
//}